internal/component/prometheus/operator/common/component_test.go

```diff
@@ -94,25 +96,25 @@ func TestRunExit(t *testing.T) {
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	cmpRunExited := atomic.Bool{}
 	cmpRunExited.Store(false)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
+		wg.Done()
 		err := c.Run(ctx)
 		require.NoError(t, err)
 		cmpRunExited.Store(true)
-		fmt.Println("component.Run exited")
 	}()
-
+	wg.Wait()
 	// Stop the component.
 	// It shouldn't stop immediately, because the CRD Manager is hung.
 	cancelFunc()
+	<-stopRun
```

Comment:
The idea behind the test is to sleep for a bit after `cancelFunc()` and make sure Run() hasn't exited, up until we call `<-stopRun`.

Reply:
Added the sleep back in.

Reply:
```suggestion
	if cmpRunExited.Load() {
		require.Fail(t, "component.Run exited")
	}

	<-stopRun
```

Thank youm but we also need the check to make sure the component didn't exit after `time.Sleep` and before `<-stopRun`.